packages/hooks/src/use-block-suite-page-meta.ts

```diff
@@ -19,30 +21,24 @@ declare module '@blocksuite/store' {
   }
 }
 
+const weakMap = new WeakMap<Workspace, Atom<PageMeta[]>>();
+
 export function useBlockSuitePageMeta(
   blockSuiteWorkspace: Workspace
 ): PageMeta[] {
-  const [pageMeta, setPageMeta] = useState<PageMeta[]>(
-    () => blockSuiteWorkspace?.meta.pageMetas ?? []
-  );
-  const [prev, setPrev] = useState(() => blockSuiteWorkspace);
-  if (prev !== blockSuiteWorkspace) {
-    setPrev(blockSuiteWorkspace);
-    if (blockSuiteWorkspace) {
-      setPageMeta(blockSuiteWorkspace.meta.pageMetas);
-    }
-  }
-  useEffect(() => {
-    if (blockSuiteWorkspace) {
+  if (!weakMap.has(blockSuiteWorkspace)) {
+    const baseAtom = atom<PageMeta[]>(blockSuiteWorkspace.meta.pageMetas);
+    weakMap.set(blockSuiteWorkspace, baseAtom);
+    baseAtom.onMount = set => {
       const dispose = blockSuiteWorkspace.meta.pageMetasUpdated.on(() => {
-        setPageMeta(blockSuiteWorkspace.meta.pageMetas);
+        set(blockSuiteWorkspace.meta.pageMetas);
       });
       return () => {
         dispose.dispose();
```

Comment:
may also need to remove the atom from map after dispose

Reply:
i think we don't, because It's weak map. the atom will be auto GC, when blocksuite workspace is deleted

Reply:
You are right, this is not actually needed